pkg/controller/sparkapplication/controller.go

```diff
@@ -318,6 +328,10 @@ func (c *Controller) getAndUpdateDriverState(app *v1beta1.SparkApplication) erro
 				app.Status.TerminationTime = currentDriverState.completionTime
 			}
 		}
+		// Expose any errors to the users.
+		if currentDriverState.err != nil {
+			app.Status.AppState.ErrorMessage += currentDriverState.err.Error()
```

Comment:
I think you can just do assign instead of appending here.

Reply:
done